File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        dictt = {}
        for n in nums:
            dictt[n] = 1
        ans = 0
        for n in nums:
            if (n-1) not in dictt.keys():
                tans = 1
                newn = n+1
                while newn in dictt.keys():
                    tans+=1
                    newn+=1
                ans = max(ans,tans)
        return ans

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int ans;

        int st = 0;
        int end = nums.size()-1;
        if(nums.size()==1){
            return nums[0];
        }
        while(st<=end){
            int mid = (st+end)/2;

            if(mid==0){
                if(nums[nums.size()-1]>nums[0]){
                    return nums[0];
                }
                else if(nums[mid]>=nums[end]){
                    st = mid+1;
                }
                else if(nums[mid]<nums[end]){
                    end = mid-1;
                }
            }
            else{
                if(nums[mid-1]>nums[mid]){
                    return nums[mid];
                }
                else if(nums[mid]>=nums[end]){
                    st = mid+1;
                }
                else if(nums[mid]<nums[end]){
                    end = mid-1;
                }
            }
        }
    }
};
